### PR TITLE
Took out JSX and added Omit to Placeholder

### LINKED ISSUE #3167: Placeholder in TextInput does not accept a react node usign typescript
Using typescript if you add a react node in the placeholder field you get this error:

```
TS2322: Type 'Element' is not assignable to type 'string | ({} & string) | (ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)> & string) | (ReactNodeArray & string) | (ReactPortal & string)'.
  Type 'Element' is not assignable to type 'ReactPortal & string'.
    Property 'children' is missing in type 'Element' but required in type 'ReactPortal'.
```

It seems there are some problems about typings in the placeholder field.

### Expected Behavior
No compiler error

### Actual Behavior

This compiler error:

```
TS2322: Type 'Element' is not assignable to type 'string | ({} & string) | (ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)> & string) | (ReactNodeArray & string) | (ReactPortal & string)'.
  Type 'Element' is not assignable to type 'ReactPortal & string'.
    Property 'children' is missing in type 'Element' but required in type 'ReactPortal'.
```

### Steps to Reproduce

Just try this:

```
<TextInput size="small" name="parcelWidth" placeholder={<Text>Foo</Text>} />
```

### Your Environment

- Grommet version: 2.7.1 with typescript v3.4.5

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR Omits Placeholder so that it can use the React.ReactNode for TS 
#### Where should the reviewer start?
docs/TextInput/index.d.ts
#### What testing has been done on this PR?
I branched out grommet with these fixes and created a react app with passing in
  <TextInput placeholder={<Text>barry</Text>}
to test for the error
#### Any background context you want to provide?
The placeholder is a HTML attribute so it was only accepting a string in which we needed to Omit and use the React.ReactNode. Also took out JSX.element since this was not the fix for this issue
#### What are the relevant issues?
closes #3167 
#### Do the grommet docs need to be updated?
auto
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
Thanks! My only concern is with removing the `element` type, I'm afraid it will cause a backward compatibility issue. Can you confirm that the prop is still type-safe even after removing it?

**Comment by britt6612** [comment]
@ShimiSun when the element type was added it never solved the issue that's how I was still able to reproduce the problem in https://codesandbox.io/s/grommet-ts-v8lyz the placeholder did not need the element to be added since it was another HTML attribute it needed to be Omit React.ReactNode covers the JSX element type

File: src/js/components/TextInput/index.d.ts [hunk 1]
```diff
@@ -14,13 +14,13 @@ export interface TextInputProps {
...
-  placeholder?: string | React.ReactNode | JSX.Element;
+  placeholder?: string | React.ReactNode;
   plain?: boolean;
   size?: "small" | "medium" | "large" | "xlarge" | string;
   suggestions?: ({label?: React.ReactNode,value?: any} | string)[];
   value?: string | number;
 }
 
-declare const TextInput: React.ComponentClass<TextInputProps & Omit<JSX.IntrinsicElements['input'], 'onSelect' | 'size'>>;
+declare const TextInput: React.ComponentClass<TextInputProps & Omit<JSX.IntrinsicElements['input'], 'onSelect' | 'size' | "placeholder">>;
```
Comment (ShimiSun) [inline src/js/components/TextInput/index.d.ts]: can you update to single quotes?

File: src/js/components/TextInput/index.d.ts [hunk 2]
```diff
@@ -14,13 +14,13 @@ export interface TextInputProps {
   onSelect?: ((x: { target: React.RefObject<HTMLElement>['current'], suggestion: any }) => void);
   onSuggestionsOpen?: ((...args: any[]) => any);
   onSuggestionsClose?: ((...args: any[]) => any);
-  placeholder?: string | React.ReactNode | JSX.Element;
+  placeholder?: string | React.ReactNode;
```
Comment (ShimiSun) [inline src/js/components/TextInput/index.d.ts]: Do we actually need the string type here?

Comment (britt6612) [reply src/js/components/TextInput/index.d.ts]: Yes now that I found about about the ReactNode we do not need string

**Pull request merged**